src/Design.cc

```diff
@@ -87,81 +92,75 @@ void Design::readDef(const std::string& file_name,
   if (tech_->getDB()->getTech() == nullptr) {
     getLogger()->error(utl::ORD, 102, "No technology has been read.");
   }
-  app->readDef(file_name.c_str(),
-               tech_->getDB()->getTech(),
-               continue_on_errors,
-               floorplan_init,
-               incremental,
-               child);
+  getOpenRoad()->readDef(file_name.c_str(),
+                         tech_->getDB()->getTech(),
+                         continue_on_errors,
+                         floorplan_init,
+                         incremental,
+                         child);
 }
 
 void Design::link(const std::string& design_name)
 {
-  auto app = OpenRoad::openRoad();
-  app->linkDesign(design_name.c_str(), false);
+  getOpenRoad()->linkDesign(design_name.c_str(), false);
 }
 
 void Design::readDb(std::istream& stream)
 {
-  auto app = OpenRoad::openRoad();
-  app->readDb(stream);
+  getOpenRoad()->readDb(stream);
 }
 
 void Design::readDb(const std::string& file_name)
 {
-  auto app = OpenRoad::openRoad();
-  app->readDb(file_name.c_str());
+  getOpenRoad()->readDb(file_name.c_str());
 }
 
 void Design::writeDb(std::ostream& stream)
 {
-  auto app = OpenRoad::openRoad();
-  app->writeDb(stream);
+  getOpenRoad()->writeDb(stream);
 }
 
 void Design::writeDb(const std::string& file_name)
 {
-  auto app = OpenRoad::openRoad();
-  app->writeDb(file_name.c_str());
+  getOpenRoad()->writeDb(file_name.c_str());
 }
 
 void Design::writeDef(const std::string& file_name)
 {
-  auto app = OpenRoad::openRoad();
-  app->writeDef(file_name.c_str(), "5.8");
+  getOpenRoad()->writeDef(file_name.c_str(), "5.8");
 }
 
-ifp::InitFloorplan* Design::getFloorplan()
+ifp::InitFloorplan Design::getFloorplan()
 {
-  auto app = OpenRoad::openRoad();
   auto block = getBlock();
   if (!block) {
     getLogger()->error(utl::ORD, 37, "No block loaded.");
   }
-  return new ifp::InitFloorplan(block, app->getLogger(), app->getDbNetwork());
+  return ifp::InitFloorplan(block, getLogger(), getSta()->getDbNetwork());
 }
 
 utl::Logger* Design::getLogger()
 {
-  auto app = OpenRoad::openRoad();
-  return app->getLogger();
+  return getOpenRoad()->getLogger();
 }
 
 int Design::micronToDBU(double coord)
 {
-  int dbuPerMicron = getBlock()->getDbUnitsPerMicron();
+  const int dbuPerMicron = getBlock()->getDbUnitsPerMicron();
   return round(coord * dbuPerMicron);
 }
 
 ant::AntennaChecker* Design::getAntennaChecker()
 {
-  auto app = OpenRoad::openRoad();
-  return app->getAntennaChecker();
+  return getOpenRoad()->getAntennaChecker();
 }
 
 const std::string Design::evalTclString(const std::string& cmd)
 {
-  Tcl_Interp* tcl_interp = OpenRoad::openRoad()->tclInterp();
+  const std::lock_guard<std::mutex> lock(interp_mutex);
+  auto openroad = getOpenRoad();
+  ord::OpenRoad::setOpenRoad(openroad, /* reinit_ok */ true);
+  Tcl_Interp* tcl_interp = openroad->tclInterp();
```

Comment:
@grotival Talked to Matt about taking this lock a while ago. If we rely on TCL scripts they will be inherently serialized for now. I think it's the right tradeoff for now otherwise it'll get a lot more complicated.